master/internal/rm/agentrm/agent_resource_manager.go

```diff
@@ -524,16 +524,34 @@ func (a *ResourceManager) TaskContainerDefaults(
 	return result, nil
 }
 
-// ValidateCommandResources implements rm.ResourceManager.
-func (a *ResourceManager) ValidateCommandResources(
-	msg sproto.ValidateCommandResourcesRequest,
-) (sproto.ValidateCommandResourcesResponse, error) {
-	pool, err := a.poolByName(msg.ResourcePool)
-	if err != nil {
-		a.syslog.WithError(err).Error("recovering job position")
-		return sproto.ValidateCommandResourcesResponse{}, err
+// ValidateResources implements rm.ResourceManager.
+func (a *ResourceManager) ValidateResources(
+	msg sproto.ValidateResourcesRequest,
+) (sproto.ValidateResourcesResponse, []command.LaunchWarning, error) {
+	if msg.Slots == 0 {
+		return sproto.ValidateResourcesResponse{}, nil, nil
+	} else if msg.IsSingleNode {
```

Comment:
for guard clauses (i.e. if-return) I'd suggest styling it as

```suggestion
	}
	
	if msg.IsSingleNode {
```

Reply:
Done!